src/main/java/core/basesyntax/services/files/FileReaderImpl.java

```
@@ -0,0 +1,33 @@
+package core.basesyntax.services.files;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class FileReaderImpl implements FileReader {
+
+    @Override
+    public List<String> read(String fileName) {
+        List<String> list = new ArrayList<>();
+        try {
+            File file = new File(fileName);
+            Scanner myReader = new Scanner(file);
+
+            if (myReader.hasNextLine()) {
+                myReader.nextLine(); // Skip the header line
+            }
+
+            while (myReader.hasNextLine()) {
+                list.add(myReader.nextLine());
+            }
+            myReader.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
```

This violates the requirement: "The program must not ignore exceptions; instead, it must rethrow them as RuntimeException with informative messages and the original exception." The code only prints the error and stack trace instead of rethrowing.